assets/queries/cloudFormation/s3_buckets_are_publicly_accessible/query.rego

```diff
@@ -0,0 +1,50 @@
+package Cx
+
+CxPolicy [ result ] {
+ 	resourcePolicy := input.document[indexPolicy].Resources[namePolicy]
+	policyProperties := [policyProperties | resourcePolicy.Type == "AWS::IAM::Policy"; policyProperties = resourcePolicy.Properties.PolicyDocument]
+	policyStatements := policyProperties[_].Statement[index]
+    
+  resourceBucket := input.document[indexBucket].Resources[nameBucket]
+  bucketProperties := [bucketProperties | resourceBucket.Type == "AWS::S3::Bucket"; bucketProperties = resourceBucket.Properties]
+
+ 	checkPolicyConfiguration(policyStatements)
+  checkPublicAccessBlockConfiguration(bucketProperties)
+  
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.Properties.PublicAccessBlockConfiguration", [nameBucket]),
+                "issueType":		"IncorrectValue",  
+                "keyExpectedValue": "'Resources.Properties.PublicAccessBlockConfiguration' is setted and configuration has value true",
+                "keyActualValue": 	"'Resources.Properties.PublicAccessBlockConfiguration' is not setted or configuration has value false "
+                }
```

Comment:
You will have to divide the query in two rules: one when "RestrictPublicBuckets" is false and another when "IgnorePublicAcls" is false.

Replace with:
```
publicAccessBlockConfiguration := resourceBucket.Properties.PublicAccessBlockConfiguration
publicAccessBlockConfiguration.RestrictPublicBuckets == false

result := {
                "documentId": 		input.document[i].id,
                "searchKey": 	    sprintf("Resources.%s.Properties.PublicAccessBlockConfigurationRestrictPublicBuckets", [nameBucket]),
                "issueType":		"IncorrectValue",  
                "keyExpectedValue": "'Resources.Properties.PublicAccessBlockConfiguration.RestrictPublicBuckets' is true",
                "keyActualValue": 	"'Resources.Properties.PublicAccessBlockConfiguration.RestrictPublicBuckets' is false"
                }
```

Then, create an equal rule for the "IgnorePublicAcls" field.

Reply:
Done